extensions/libraries/redcore/api/hal/hal.php

```diff
@@ -231,6 +231,12 @@ public function setOptionsFromHeader()
 			RBootstrap::$config->set('enable_translation_fallback_webservices', $fallbackEnabled);
 		}
 
+		if (isset($headers['X_WEBSERVICE_STATEFUL']))
+		{
+			$useState = (int) $headers['X_WEBSERVICE_STATEFUL'] == 1 ? 1 : 0;
+			$this->options->set('webservice_stateful', $useState);
```

Comment:
you should also add a set config here:
RBootstrap::$config->set('webservices_stateful', $useState);